### DISCUSSION TITLE
Database base Pushing table issue

### DISCUSSION THREAD

**Discussion by Yahyakhan1048** [prisma/prisma#22316] [Q&A] [answered]
### Question

I found this issue when pushing table from next.js application to postgreSql database
```
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "aws-0-us-west-1.pooler.supabase.com:6543"

Error: Schema engine error:
ERROR: prepared statement "s0" does not exist
while my database url is this one
DATABASE_URL="postgres://postgres.pxzfdiupwnecjzoriwwc:(password)@aws-0-us-west-1.pooler.supabase.com:6543/postgres"
and this is the code inside the schema-prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Yahyakhan1048 👋 

Thank you for raising this question.

1. Is your application a long-lived one (i.e connecting to a database and maintaining a connection)? if so, you should use the direct connection with the port `5432`. 

2. Are you connecting to your database and then disconnecting immediately (e.g. a serverless environment)? if so, you should use the pooled connection with port `6543` and you may need to enable PgBouncer mode by adding `&pgbouncer=true` to it. i.e 

```
postgres://postgres.pxzfdiupwnecjzoriwwc:(password)@aws-0-us-west-1.pooler.supabase.com:6543/postgres&pgbouncer=true`
```

You can also take a look at this [comment](https://github.com/prisma/prisma/issues/11831#issuecomment-1825914088) in this issue which discusses a similar problem as you're facing.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Yahyakhan1048** [reply]
ok thanks bro i will apply them then i will inform you

**Reply by mpkrueger** [reply]
Switching to 5432 did the trick for me in this instance.

**Reply by ludralph** [reply, marked as answer]
Hi @Yahyakhan1048 

do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!
